src/react/reconcilation.js

```diff
@@ -1476,4 +1482,26 @@ export class Reconciler {
       }
     }
   }
+
+  _handleReportedSideEffect(
+    sideEffectType: SideEffectType,
+    binding: void | Binding | PropertyBinding,
+    value: void | Value
+  ): void {
+    let location = getLocationFromValue(value);
+
+    if (sideEffectType === "MODIFIED_BINDING") {
+      let name = binding ? `"${((binding: any): Binding).name}"` : "unknown";
+      throw new UnsupportedSideEffect(`side-effects from mutating the binding ${name}${location}`);
+    } else if (sideEffectType === "MODIFIED_PROPERTY") {
+      let name = "";
+      let key = ((binding: any): PropertyBinding).key;
+      if (typeof key === "string") {
+        name = `"${key}"`;
+      }
+      throw new UnsupportedSideEffect(`side-effects from mutating the object ${name}${location}`);
```

Comment:
"from mutating a property" maybe?

I'm seeing an error like

>Failed to render React component "Tooltip" due to side-effects from mutating the object "(\d)(\d{3})($|\D)"

when it points to code like

```js
var _regexCache = {};
```

I assume the `name` is the property name here, not the object name.